api/query/cache/poll/poll.go

```diff
@@ -84,20 +87,77 @@ func wait(start time.Time, total time.Duration) {
 	}
 }
 
-// KeepMetadataUpdated implements updates to metadataMapCached via simple polling every
+// MetadataPollingService performs metadata-related services via simple polling every
 // interval duration.
-func KeepMetadataUpdated(client *http.Client, logger shared.Logger, interval time.Duration) {
-	logger.Infof("Metadata update via polling started")
+func MetadataPollingService(ctx context.Context, logger shared.Logger, interval time.Duration) {
+	logger.Infof("Metadata polling service started")
+	netClient := &http.Client{Timeout: time.Second * 5}
+	cacheSet := shared.NewRedisSet()
+	gitHubClient, err := shared.NewAppEngineAPI(ctx).GetGitHubClient()
+	if err != nil {
+		logger.Infof("Unable to get GitHub client: %v", err)
+	}
+
 	for {
-		metadataCache, err := shared.GetWPTMetadataArchive(client, nil)
+		keepMetadataUpdated(netClient, logger)
+		cleanOrphanedPendingMetadata(ctx, gitHubClient, cacheSet, logger)
+		time.Sleep(interval)
+	}
+}
+
+// keepMetadataUpdated fetches a new copy of the wpt-metadata repo and updates metadataMapCached.
+func keepMetadataUpdated(client *http.Client, logger shared.Logger) {
+	logger.Infof("Running keepMetadataUpdated...")
+	metadataCache, err := shared.GetWPTMetadataArchive(client, nil)
+	if err != nil {
+		logger.Errorf("Error fetching Metadata for update: %v", err)
+		return
+	}
+
+	if metadataCache != nil {
+		query.MetadataMapCached = metadataCache
+	}
+}
+
+// cleanOrphanedPendingMetadata cleans and removes orphaned pending metadata in Redis.
+func cleanOrphanedPendingMetadata(ctx context.Context, ghClient *github.Client, cacheSet shared.RedisSet, logger shared.Logger) {
+	logger.Infof("Running cleanOrphanedPendingMetadata...")
+	if ghClient == nil {
```

Comment:
Optional: perhaps we should skip calling the function in the outer-loop, and require the ghClient be non-null?